File: frontend/src/app/core/models/order.ts
export enum OrderStatus {
  Confirmed = 'Confirmed',
  Processing = 'Processing',
  Shipped = 'Shipped',
  Delivered = 'Delivered',
}

export interface OrderItem {
  productId: number;
  name: string;
  price: number;
  quantity: number;
  color: string;
  size: string;
  imageUrl: string;
  imageAlt: string;
  sku: string;
}

export interface CustomerDetails {
  name: string;
  phone: string;
  address: string;
  deliveryDetails: string;
}

export interface OrderTotals {
  subtotal: number;
  shipping: number;
  tax: number;
  total: number;
}

export interface OrderTimeline {
  confirmedDate: string;
  processingLabel: string;
  shippedEta: string;
  deliveredEta: string;
}

export interface Order {
  id: string;
  status: OrderStatus;
  items: OrderItem[];
  customer: CustomerDetails;
  totals: OrderTotals;
  timeline: OrderTimeline;
}

File: backend/src/ECommerce.Core/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace ECommerce.Core.Entities;

public enum OrderStatus
{
    Pending,
    Confirmed,
    Processing,
    Shipped,
    Delivered,
    Cancelled
}

public class Order : BaseEntity
{
    public string OrderNumber { get; set; } // e.g. ORD-2024-001
    
    // Customer Info (Snapshot)
    public string CustomerName { get; set; }
    public string CustomerPhone { get; set; }
    public string ShippingAddress { get; set; }
    public string? DeliveryDetails { get; set; }
    
    // Financials
    public decimal SubTotal { get; set; }
    public decimal Tax { get; set; }
    public decimal ShippingCost { get; set; }
    public decimal Total { get; set; }
    
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

public class OrderItem : BaseEntity
{
    public int OrderId { get; set; }
    public Order Order { get; set; }
    
    public int ProductId { get; set; }
    public Product Product { get; set; }
    
    public string ProductName { get; set; }
    public string? Color { get; set; }
    public string? Size { get; set; }
    
    public decimal UnitPrice { get; set; }
    public int Quantity { get; set; }
    public decimal TotalPrice => UnitPrice * Quantity;
}
